docusaurus/docs/Android/02-migration-guides/01-client/02-result.mdx

```diff
@@ -0,0 +1,101 @@
+![](../../assets/railway-oriented-programming.jpg)
+
+## Result
+
+This is a basic model to represent a normalized result from business work. This looks similar to [Kotlin's Result](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-result/), but Stream Result was designed to include more information about success and error and support more convenient functionalities to handle results. Result is basically consist of two detailed types below:
+
+- **`Result.Success`**: This represents your business's successful result, including a `value` property, a generic type of Result.
+- **`Result.Failure`**: This represents the failed result of your business result and includes a `value` property, the `Error` type.
+
+The old way to work with `result` was by checking if the result was successful or not and perform different actions on any case, something like:
+```kotlin {2,3,6}
+val result = obtainResult()
+if (result.isSuccess) {
+    val data = result.data()
+    [...] // work with successful data
+} else {
+    val error = result.error()
+    [...] // Notify an error happened
+}
+```
+
+On the new implementation it has been improved, and allows you to use it in an easier and safer way:
+```kotlin
+val result = obtainResult()
+result.onSuccess { data ->
+    [...] // work with successful data
+}.onError { error ->
+    [...] // Notify an error happened
+}
+```
+
+## Error
+
+`Result.Failure` has `Error` as a value property, which contains error details of your business work. Basically, `Error` consists of three different types of errors below:
+
+- **`Error.GenericError`**: Represents a normal type of error and only contains an error message.
+- **`Error.ThrowableError`**: Represents an exceptional type of error and contains a message and cause information.
+- **`Error.NetworkError`**: Represents a network error and contains status code, message, and cause information.
+
+## Result Extensions
+
+**Stream Result** library useful extensions below to effectively achieve Railway Oriented Programming in Kotlin:
+
+### Result.then
+
+Composition the `Result` with a given `Result` from a lambda function.
+
+```kotlin
+val result0: Result<String> = Result.Success(value = "result0")
+val result1: Result<Int> = Result.Success(value = 123)
+val result = result0 then { result1 }
+result.onSuccess { intValue -> .. }
+```
+
+### Result.map, Result.mapSuspend
+
+Returns a transformed `Result` of applying the given function if the `Result` contains a successful data payload.
+
+```kotlin
+val result: Result<String> = Result.Success(value = "result")
+val mappedResult = result.map { 123 }
+mappedResult.onSuccess { intValue -> }
+```
+
+### Result.flatMap, Result.flatMapSuspend
+
+Returns a transformed `Result` from results of the function if the `Result` contains a successful data payload. Returns an original `Result` if the `Result` contains an error payload.
```

Comment:
```suggestion
Returns a transformed `Result` obtained from the results of the function if the `Result` contains a successful data payload. Otherwise, returns the original `Result` if the `Result` contains an error payload.
```